src/parser/mod.rs

```diff
@@ -10408,13 +10408,86 @@ impl<'a> Parser<'a> {
             None
         };
 
+        let with_fill = if dialect_of!(self is ClickHouseDialect | GenericDialect)
+            && self.parse_keywords(&[Keyword::WITH, Keyword::FILL])
+        {
+            Some(self.parse_with_fill()?)
+        } else {
+            None
+        };
+
+        let interpolate = if dialect_of!(self is ClickHouseDialect | GenericDialect)
+            && self.parse_keyword(Keyword::INTERPOLATE)
+        {
+            if self.consume_token(&Token::LParen) {
+                if self.peek_token().token == Token::RParen {
```

Comment:
```suggestion
                if self.consume_token(&Token::RParen) {
```
this could do similar to lparen?

Reply:
Will swap to use `parse_comma_separated0` so we don't need this at all.